File: 1046-max-consecutive-ones-iii/1046-max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int res = 0;
        int left = 0;
        int skip = 0;
        for (int right = 0; right < nums.length; right++) {
            if (nums[right] == 0) {
                skip++;
            }
            while (skip > k) {
                if (nums[left] == 0) {
                    skip--;
                }
                left++;
            }
            res = Math.max(right - left + 1, res);
        }
        return res;
    }
}

File: 1046-max-consecutive-ones-iii/1046-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        res = 0
        skip = 0
        start = 0
        for end in range(len(nums)):
            if nums[end] == 0:
                skip += 1
            while skip > k:
                if nums[start] == 0:
                    skip -= 1
                start += 1
            res = max(res, end - start + 1)
        return res